### PR TITLE
DEV: Keep global state reset in a single place

### PR DESCRIPTION
**Pull request by CvX**

### CODE REVIEW

File: app/assets/javascripts/discourse/tests/acceptance/sidebar-plugin-api-test.js [hunk 1]
```diff
@@ -21,7 +19,6 @@ acceptance("Sidebar - Plugin API", function (needs) {
   });
 
   needs.hooks.afterEach(() => {
-    resetSidebarSection();
```
Comment (davidtaylorhq) [inline app/assets/javascripts/discourse/tests/acceptance/sidebar-plugin-api-test.js:24]: Doesn't look like this has been introduced elsewhere... is it not needed at all?

File: app/assets/javascripts/discourse/tests/acceptance/plugin-outlet-multi-template-test.js [hunk 2]
```diff
@@ -26,7 +24,6 @@ acceptance("Plugin Outlet - Multi Template", function (needs) {
     delete Ember.TEMPLATES[HELLO];
     // eslint-disable-next-line no-undef
     delete Ember.TEMPLATES[GOODBYE];
-    clearCache();
```
Comment (davidtaylorhq) [inline app/assets/javascripts/discourse/tests/acceptance/plugin-outlet-multi-template-test.js:29]: Doesn't look like this has been introduced elsewhere... is it not needed at all?

File: app/assets/javascripts/discourse/tests/acceptance/sidebar-plugin-api-test.js (hunk 1 above)
Comment (CvX) [reply app/assets/javascripts/discourse/tests/acceptance/sidebar-plugin-api-test.js:24]: It already being done after each test:

https://github.com/discourse/discourse/blob/e21d2ac217e146005d8454b08f2de1e43d6dc3de/app/assets/javascripts/discourse/tests/helpers/qunit-helpers.js#L203-L203

File: app/assets/javascripts/discourse/tests/acceptance/plugin-outlet-multi-template-test.js (hunk 2 above)
Comment (CvX) [reply app/assets/javascripts/discourse/tests/acceptance/plugin-outlet-multi-template-test.js:29]: Also being done after each test:

https://github.com/discourse/discourse/blob/e21d2ac217e146005d8454b08f2de1e43d6dc3de/app/assets/javascripts/discourse/tests/helpers/qunit-helpers.js#L163-L163

**Pull request merged**